src/hyppo/dos.asm

```diff
@@ -2603,6 +2603,25 @@ drce_cont2:
         ;;
         and #$06                ;; %00000110
         beq drce_cont3        ;; branch if equal to zero (ie not Hidden OR System)
+
+        ;; macOS creates the `.` and `..` entries as hidden. Special case `..`
+        ;; so that it appears in listings. Some uses, like the Freezer, need it
+        ;; visible to know how to navigate to the parent directory.
+
+        bra +
+dotdotshortname:
+        !text "..         "
++       ldx #11
+        ldy #fs_fat32_dirent_offset_shortname        ;; Y=0 (first char of entry)
+-       lda (<dos_scratch_vector),y
+        cmp dotdotshortname,y
+        bne drce_cont2b
+        iny
+        dex
+        bne -
+        bra drce_cont3
```

Comment:
Looks like I can simplify this by moving the `jmp drce_normalrecord` up here, if that's preferred. I was trying to preserve the ordering.

Reply:
Yes, sounds good!